### DISCUSSION TITLE
Using library with AddDbContextPool

### DISCUSSION THREAD

**Discussion by Hantick** [Giorgi/EntityFramework.Exceptions#60] [Q&A] [answered]
Hi, is it possible to use this library with `AddDbContextPool` method on startup? It gives me
 `''OnConfiguring' cannot be used to modify DbContextOptions when DbContext pooling is enabled.'`
 when trying to use it with pooling

**Comment by Giorgi** [comment]
Did you read the docs for [DbContext Pooling](https://learn.microsoft.com/en-us/ef/core/performance/advanced-performance-topics?tabs=without-di%2Cexpression-api-with-constant#dbcontext-pooling)?

**Comment by Hantick** [comment]
> Did you read the docs for [DbContext Pooling](https://learn.microsoft.com/en-us/ef/core/performance/advanced-performance-topics?tabs=without-di%2Cexpression-api-with-constant#dbcontext-pooling)?

Yes

**Comment by Giorgi** [comment]
Can you post your code?

**Comment by Hantick** [comment]
Program.cs
```
services.AddDbContextPool<ApplicationDbContext>(opt => opt.UseNpgsql(connectionString));
services.AddDatabaseDeveloperPageExceptionFilter();
```

ApplicationDbContext.cs
```
 public class ApplicationDbContext : IdentityDbContext
     <UserModel, RoleModel, string, IdentityUserClaim<string>, UserRoleModel, IdentityUserLogin<string>, IdentityRoleClaim<string>, IdentityUserToken<string>>
 {
      // Db sets

     public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options) : base(options) { }

     protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) => optionsBuilder.UseExceptionProcessor();

     protected override void OnModelCreating(ModelBuilder modelBuilder)
     {
         // Stuff
     }
```

Exception I get:
`System.InvalidOperationException: ''OnConfiguring' cannot be used to modify DbContextOptions when DbContext pooling is enabled.'`

**Reply by Giorgi** [reply, marked as answer]
You need to move `UseExceptionProcessor` after `UseNpgsql`. That's where you configure the `DbContextPool`
